tripal_chado/tests/src/Kernel/Plugin/ChadoBuddy/ChadoCvtermBuddyTest.php

```diff
@@ -13,31 +13,92 @@
 class ChadoCvtermBuddyTest extends ChadoTestKernelBase {
   protected $defaultTheme = 'stark';
 
+  protected ChadoConnection $connection;
+
   protected static $modules = ['system', 'user', 'file', 'tripal', 'tripal_chado'];
 
-  protected ChadoConnection $connection;
+  /**
+   * {@inheritdoc}
+   */
+  protected function setUp(): void {
+    parent::setUp();
+
+    // Open connection to a test Chado
+    $this->connection = $this->getTestSchema(ChadoTestKernelBase::INIT_DUMMY);
```

Comment:
It would be good to add our own cvterm-focused test chado instance rather then re-using this non-standard one added by Valentin.

I would suggest adding a new fixture `cvterm_focused_schema.sql` with an SQL comment at the top indicating which tables it includes and that it includes test data. Make it clear these table definitions match Chado 1.3 and that this schema can be used when testing cvterm-related functionality.

Then to make this work within our automated testing you would:
1. Add a block like this to the chado test trait: https://github.com/tripal/tripal/blob/4.x/tripal_chado/tests/src/Traits/ChadoTestTrait.php#L319-L328
2. Add the same constant to the functional and kernel test bases like this: https://github.com/tripal/tripal/blob/4.x/tripal_chado/tests/src/Traits/ChadoTestTrait.php#L319-L328 and https://github.com/tripal/tripal/blob/4.x/tripal_chado/tests/src/Traits/ChadoTestTrait.php#L319-L328

A good constant might be `CHADO_CVTERM_SUBSET`. For linked tables you could keep feature or move to something simpler like contact or protocol which have a type_id but no other FKs.

Reply:
I am seeing about 4.7 seconds to create a test chado with
`$this->connection = $this->getTestSchema(ChadoTestKernelBase::PREPARE_TEST_CHADO);`
vs. 0.3 seconds for the `INIT_DUMMY` schema.

So, I am back to using `PREPARE_TEST_CHADO`, because the super-simplified `INIT_DUMMY` schema does not allow looking up table columns or primary key. Making a new "faster" test schema can be a future project.

Reply:
This is moved to issue #1947 so I will resolve the conversation here